read.go

```diff
@@ -459,6 +459,110 @@ func readFetchResponseHeaderV5(r *bufio.Reader, size int) (throttle int32, water
 
 }
 
+func readFetchResponseHeaderV10(r *bufio.Reader, size int) (throttle int32, watermark int64, remain int, err error) {
```

Comment:
I worry a bit about the amount of copy/paste happening here.  Not sure that I have an answer now for how to fix it, but I'd worry about bugfixes making it to all the right places.

Reply:
Yes, I'm also pretty worried about this getting hard to maintain as well. Especially if we keep adding more and more versions.

Reply:
We could find a way to factorize this code but I'm worried about adding too much complexity since the Kafka protocol can add fields in the nested structures.